### ISSUE TITLE
508-defect-2: Issue checkboxes's (a) descriptions don't read out and (b) titles are verbose

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#54627]
### Point of contact

joshkimux

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

Currently, the issue checkboxes are verbose due to:
- repeating the name of the issue (as part of the label, as `sr-only` text, and as a heading)
- being coded as a description list, on iOS "description" is read out after each string 

There are also issues around the screen reader navigation experience in forms mode, as tabbing through the checkboxes only  provides context to the titles (without the descriptions).

On NVDA, it reads as:
- [Tab] Select the issues you'd like for us to review, grouping list with 2 items, Ankylosis of knee check box not checked
- [Down arrow] Ankylosis of knee
- [Down arrow] Heading level 4 Ankylosis of knee
- [Down arrow] Service connection for Ankylosis of knee...
- [Down arrow] Current rating: 10% 
- [Down arrow] Decision date: March 12, 2022

On iOS, it reads as
- [Swipe] Ankylosis of knee, Checkbox, checked, description list start
- [Swipe] Ankylosis of knee, heading level 4, description
- [Swipe] Service connection for Ankylosis of knee... description
- [Swipe] Current rating, description
- [Swipe] 10, description
- [Swipe] Percent, description

https://user-images.githubusercontent.com/14154792/223213582-a81d8654-3beb-462a-ae54-8b66f8300952.MP4

### Reproduction steps

1) Open up VoiceOver (or any screen reader) on the contestable issue page
2) Tab to the first checkbox
3) Confirm that "Ankylosis of Knee" reads out, but the rest of the description is absent
4) Navigate line by line through the issue
5) Confirm that the announcements append "description" to each line

### Proposed solution or next steps

Ideally, we should likely do the following:
- [ ] Add an `aria-labelledby` to the checkbox's label to point to the visible `h4`, instead of using a `sr-only` label
- [ ] Add an `aria-describedby` to the checkbox itself to use the contents of "div class="widget-content-wrap" to describe the checkbox
- [ ] Remove `dt` and `dd` elements in favor of just a `h4` with `p` pairing
- [ ] Nest each of the checkbox sections in an unordered list

For example:
```html
<!-- Convert to unordered list, not a description list -->
<ul>

<!-- Convert to list item -->
  <li id="issue-0" style="list-style: none;">
  
    <!-- Contains the checkbox -->
    <div> 

      <!-- Remove `sr-only`, add aria-describedby to input, point to the first description -->
      <input type="checkbox" id="root_contestedIssues_0" name="root_contestedIssues_0" aria-describedby="desc-1">

      <!-- Add aria-labelledby to empty label, point to first `h4`  -->
      <label class="schemaform-label" for="root_contestedIssues_0" aria-labelledby="title-1"></label>   
    </div> 
  
    <!-- Contains the title/description -->
    <div data-index="0">

      <!-- Add unique id to be referenced by aria-labelledby -->
      <h4 id="title-1">Ankylosis of knee</h4>

      <!-- Add unique id to be referenced by aria-describedby -->
      <div id="desc-1">
        <p>Service connection for Ankylosis of knee is granted with an evaluation of 10% effective December 2, 2020.</p>
        <p>Current rating: <strong>10%</strong></p>
        <p>Decision date: <strong>March 12, 2022</strong></p>
      </div>
    </div>
  
  </li>
</ul>
```

### References, articles, or WCAG support

1.
2.
3.
...

### Type of issue

- [X] Screenreader
- [ ] Keyboard
- [ ] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by Mottie** [comment]
You didn't mention this in the ticket, but after editing the issue focus is placed on the edit link (after canceling) or the issue checkbox (after updating). And on the add a new issue action link after removing an issue

![three issues shown on review & submit issues page. Edit chapter and focus moves to the first checkbox. Editing the second entry goes to the edit issue page, upon canceling, the focus is returned to the edit button on the review & submit issue. Editing again, then updating the issue returns to the issus checkbox on the review & submit page. After removing the third issue, focus is sent to the add a new issue action link](https://user-images.githubusercontent.com/136959/225440473-5faea682-a146-4fa4-ba9a-d209ef18a749.gif)

**Comment by Mottie** [comment]
Work merged in, but not yet available in staging.

**Comment by joshkimux** [comment]
### all the `defect-2`s are addressed 😄 

<img width="603" alt="Screenshot 2023-03-20 at 9 17 57 AM" src="https://user-images.githubusercontent.com/14154792/226350776-40c271ce-7311-425e-954d-ba90710d0313.png">

### I made a mistake in my original recommendation for the `defect-3` 😢 
In my original code recommendation, I made the following mistakes:
- attached `aria-labelledby` to the `label` element, when it should be on the input itself
- did not make it clear that the `sr-only` should be deleted

To fix this, we can do the following:
- [ ] move the `aria-labelledby` to the `input`, not the `label` element
- [ ] delete the `aria-labelledby` off the `label` element
- [ ] delete the `sr-only` `span` within the `label` (effectively we'd have an empty label just for the `:before` to show the checkbox (I'm guessing that's how the code works?))
- [ ] move the `aria-describedby` div so it _doesn't_ wrap the `h4` to avoid it repeating the title in the description again

In the code, it would look like this (simplified):

```html
<!-- move aria-labellebedby to input from the label -->
<input type="checkbox" id="root_contestedIssues_0" name="root_contestedIssues_0" aria-labelledby="title-1" aria-describedby="desc-1">

<!-- delete aria-labelledby from label, as we don't want to relabel a label  -->
<label class="schemaform-label" for="root_contestedIssues_0">
  :before 
  <!-- Delete `sr-only` `span` from this label, as it's effectively just providing the checkbox -->
</label>   

<h4 id="title-1">Ankylosis of knee</h4>

<!-- Ensure the aria describedby div wraps the description, but _not_ the `h4` to avoid repeating the title -->
<div id="desc-1">
  <p>Service connection for Ankylosis of knee is granted with an evaluation of 10% effective December 2, 2020.</p>
  <p>Current rating: <strong>10%</strong></p>
  <p>Decision date: <strong>March 12, 2022</strong></p>
</div>


### Comparing the behavior

https://user-images.githubusercontent.com/14154792/226356742-39ab81af-ce22-45bf-893b-666ee7eb440b.mov

Here's a recording of the above fixes comparing how it reads out before (Ankylosis of Knee) vs. after (PTSD)
- (Good): Ankylosis of Knee is announced only _once_ and is not repeated in its aria description
- (Needs some love): PTSD is repeated three times (label, `sr-only`, description)

**Comment by Mottie** [comment]
Back to you @joshkimux - ready for testing in staging
